core/controllers/classifier_test.py

```diff
@@ -143,11 +145,32 @@ def setUp(self) -> None:
                     'utf-8'),
                 self.payload_for_fetching_next_job_request['vm_id']))
 
+    def _mock_get_secret(self, name: str) -> Optional[str]:
+        """Mock for the get_secret function.
+
+        Args:
+            name: str. The name of the secret to retrieve the value.
+
+        Returns:
+            Optional[str]. The value of the secret.
+        """
+        if name == 'VM_ID':
+            return 'vm_default'
+        elif name == 'SHARED_SECRET_KEY':
+            return '1a2b3c4e'
+        return None
+
     def test_trained_classifier_handler(self) -> None:
         # Normal end-to-end test.
-        self.post_blob(
```

Comment:
Can you explain the need for the changes in this file -- why do we need the mock (or the secret values) now when we didn't before?

Reply:
Sure, So the `get_secret` function gets used in a decorator from the `acl_decorators` file which is `is_from_oppia_ml`. Now this decorator gets used in 2 handlers which are `TrainedClassifierHandler` and `NextJobHandler`. The mock changes that are recently added are part of these handlers.
I think I might've missed adding the tests before as I was adding several other tests at other places.

Reply:
Yup, but why didn't these tests fail before?

Reply:
They were failing before as well. As we have a bunch of backend tests and if any of the test fail from the queue then the rest of the backend tests gets canceled, that is why I think these were not reported before, and as I was fixing the tests the rest got reported.
If this is what you are asking here :).

Thanks

Reply:
No, I meant, why aren't they failing in develop?

Reply:
I and Sean discussed this in the data-stability team meeting. Just to update the thread, Previously the reason why we did not require to mock the secrets was that we were using the vmid-secrets config properties which were returning the default value at that time and now we have removed the config property and decided to go ahead with the cloud secrets and that is why we require to mock the tests here as part of the development server.

Thanks